include/arch/structs.h

```diff
@@ -31,6 +31,12 @@
 
 /* Per CPU architecture specifics (empty) */
 struct _cpu_arch {
+#ifdef __cplusplus
+	/* This struct will have a size 0 in C which is not allowed in C++ (it'll have a size 1). To
+	 * prevent this, we add a 1 byte dummy variable.
+	 */
+	uint8_t dummy;
+#endif
```

Comment:
Please pull the changes in `include/` into a separate commit

Reply:
Done